src/elements/title/index.js

```diff
@@ -24,12 +42,18 @@ const Title = forwardRef( ( {
 	className,
 	...props
 }, ref ) => {
+	
+	// If `as` is default and `size` is set, update Component to proper heading level
+	if ( Component === 'h1' && typeof size !== undefined ) {
```

Comment:
Would the following case be valid?

```jsx
<Title as="h1" size={3}>Example title</Title>
```

In this example, the user wants an h1 to be styled as a (presumably) smaller component, but to keep the semantic markup weight of an `h1`. I believe in this case, the Component would be re-rendered to `h3` without a way to override it.

Perhaps one option to retain this backwards compatibility would be to set the defaultProps for `as` to `undefined`, and check for `Component === undefined` here and later defining a default when this block does not apply.